srcpkgs/wireplumber/files/wireplumber/run

```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+# this service is experimental and most setups should start pipewire and wireplumber as a user,
+# for further information, please refer to the handbook
+sv check pipewire > /dev/null || exit 1
+exec env HOME=/var/run/pipewire chpst -u _pipewire:_pipewire:audio:video dbus-run-session wireplumber
```

Comment:
What does wireplumber do with a dbus user session? Using `dbus-run-session` makes a new, isolated bus that will be useless for external processes. Also, does a dbus user session require the existence of a system bus? You may need to `sv check dbus` to make sure the system bus is up. (I'm not sure about this.)

Reply:
Yep, under the virtual `_pipewire` user for global access, like pipewire global service does.

> You may need to `sv check dbus` to make sure the system bus is up.
Yup, I forgot about that, sorry.

Reply:
The dbus system bus is not exactly needed by the wireplumber actually, only some stuffs like bluetooth receiver (e.g. from mobile) I see is not working, which works after enabling dbus. Should it be mandatory to have it checked?

Reply:
Note that the issue with `dbus-run-session` isn't that it makes a bus specific to `_pipewire`, but that it creates a *random* bus that *no other process*---including pipewire---will know about. Using `dbus-run-session` in, say, your xinitrc works as expected because a random bus is created and then all processes inherit an environment variable telling them where to find the bus.

If any process needs to communicate with wireplumber via this dbus session, dbus will have to be configured (as will all who wish to talk to it) to use a well-known socket. If nothing actually talks to wireplumber on this bus, why does wireplumber require it?

Reply:
I'm not too sure why its necessary, I just created an upstream issue [wireplumber#237](https://gitlab.freedesktop.org/pipewire/wireplumber/-/issues/237) for the enquiry.

Reply:
@Animeshz Wireplumber uses the dbus bus to connect to pipewire, IIRC

Reply:
https://wiki.alpinelinux.org/wiki/PipeWire#D-Bus